templates/openssl.cnf.erb

```diff
@@ -9,7 +9,7 @@ engines                 = engine_section
 [ ca ]
 default_ca	= CA_default
 [ CA_default ]
-dir		= <%= node['openvpn']['key_dir'] %>
+dir		= $ENV::KEY_DIR
```

Comment:
This feels wrong, we should be just accessing the vars directly, not using envs?

Reply:
Setting the vars directly during static file generation seems appropriate if you'd like a cleaner, more universal openssl.cnf. However, as far as I can tell, this limits the configurability of user certificates. Previously, this resource was using envs to set the Common Name. Now, more envs are used so the resource can set more key information per call, although I could understand not needing this for key_dir. The use of envs specifically is inspired by pkitool's openssl cnf linked [here](https://github.com/OpenVPN/easy-rsa-old/blob/master/easy-rsa/2.0/openssl-1.0.0.cnf). The cookbook attributes are still used for configuration defaults, however, anyone using the user resource can now override them. Please let me know if adding envs to allow for more customization of user certificates isn't appropriate here.

Reply:
I think it is best to ask @sous-chefs/board to weigh in here

Reply:
I think it's OK to use vars here, maybe not for this one specifically but the others are OK IMO.